streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -82,6 +82,34 @@ abstract class ZStream[-R, +E, +O](
   ): ZStream[R1, E1, O1] =
     ZStream.absolve(ev(self))
 
+  /**
+   * Applies an aggregator to the stream, which converts one or more elements
+   * of type `A` into elements of type `B`.
+   */
+  def aggregate[R1 <: R, E1 >: E, O1 >: O, P](sink: ZTransducer[R1, E1, O1, P]): ZStream[R1, E1, P] =
+    ZStream {
+      for {
+        pull <- self.process
+        push <- sink.push
+        done <- ZRef.makeManaged(false)
+        run = {
+          def go: ZIO[R1, Option[E1], Chunk[P]] = done.get.flatMap {
+            if (_)
+              push(None)
```

Comment:
I'm still not convinced this is correct because in the transducer failing with `None` means give me more, whereas failing with `None` in the stream means this is the end.

Reply:
Oh wait I think I am confused. Transducer failing with `None` should mean `end`, not give me more. A transducer can just send an empty Chunk if it doesn't have anything to send.

Reply:
Ha. Ok so that changes everything 😆 give me a sec.